base/applications/rapps/cabinet.cpp

```diff
@@ -0,0 +1,351 @@
+
+    if (WriteFile((HANDLE) hf, pv, cb, &dwBytesWritten, NULL) == FALSE)
+    {
+        dwBytesWritten = (DWORD) -1;
+    }
+
+    return dwBytesWritten;
+}
+
+FNCLOSE(fnFileClose)
+{
+    return (CloseHandle((HANDLE) hf) != FALSE) ? 0 : -1;
+}
+
+FNSEEK(fnFileSeek)
+{
+    return SetFilePointer((HANDLE) hf, dist, NULL, seektype);
+}
+
+/* FDICopy callbacks */
+
+FNFDINOTIFY(fnNotify)
+{
+    INT_PTR iResult = 0;
+
+    switch (fdint)
+    {
+    case fdintCOPY_FILE:
+    {
+        ATL::CStringW szNewFileName, szExtractDir, szCabFileName;
+        ATL::CStringA szFilePathUTF8;
+
+        // Append the destination directory to the file name.
+        UTF8PathToWide((LPCSTR) pfdin->pv, szExtractDir);
+        ANSIPathToWide(pfdin->psz1, szCabFileName);
+
+        szNewFileName = szExtractDir + L"\\" + szCabFileName;
+
+        WidePathToUTF8(szNewFileName, szFilePathUTF8);
+
+        // Copy file
+        iResult = fnFileOpen((LPSTR) szFilePathUTF8.GetString(), 
+                             _O_WRONLY | _O_CREAT,
+                             0);
+    }
+    break;
+    case fdintCLOSE_FILE_INFO:
+        // Ignore all props and just close the file
+        iResult = !fnFileClose(pfdin->hf) ? TRUE : FALSE;
+        break;
+    case fdintNEXT_CABINET:
+        if (pfdin->fdie != FDIERROR_NONE)
+        {
+            iResult = -1;
+        }
+        break;
+    case fdintPARTIAL_FILE:
+        iResult = 0;
+        break;
+    case fdintCABINET_INFO:
+        iResult = 0;
+        break;
+    case fdintENUMERATE:
+        iResult = 0;
+        break;
+    default:
+        iResult = -1;
+        break;
+    }
+
+    return iResult;
+}
+
+/* cabinet.dll FDI function pointers */
+
+typedef HFDI(*fnFDICreate)(PFNALLOC, 
+                           PFNFREE, 
+                           PFNOPEN, 
+                           PFNREAD, 
+                           PFNWRITE,
+                           PFNCLOSE, 
+                           PFNSEEK, 
+                           int, 
+                           PERF);
+
+typedef BOOL(*fnFDICopy)(HFDI,
+                         LPSTR,
+                         LPSTR,
+                         INT,
+                         PFNFDINOTIFY,
+                         PFNFDIDECRYPT,
+                         void FAR *pvUser);
+
+typedef BOOL(*fnFDIDestroy)(HFDI);
+
+/* 
+ * Extraction function 
+ * TODO: require only a full path to the cab as an argument
+ */
+BOOL ExtractFilesFromCab(const ATL::CStringW& szCabName, 
+                         const ATL::CStringW& szCabDir, 
+                         const ATL::CStringW& szOutputDir)
+{
+    HINSTANCE hCabinetDll;
+    HFDI ExtractHandler;
+    ERF ExtractErrors;
+    ATL::CStringA szCabNameUTF8, szCabDirUTF8, szOutputDirUTF8;
+    fnFDICreate pfnFDICreate;
+    fnFDICopy pfnFDICopy;
+    fnFDIDestroy pfnFDIDestroy;
+    BOOL bResult;
+
+    // Load cabinet.dll and extract needed functions 
+    hCabinetDll = LoadLibraryW(L"cabinet.dll");
```

Comment:
You don't link statically to cabinet.dll?

Reply:
No, and it didn't before the change.